extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java

```diff
@@ -197,4 +208,90 @@ public boolean isAnyPropertySet() {
             return sql || jdbcWarnings.isPresent();
         }
     }
+
+    @ConfigGroup
+    public static class HibernateOrmConfigCache {
+
+        /**
+         * Cache configuration properties
+         */
+        @ConfigItem(name = ConfigItem.PARENT)
+        public Map<String, CacheConfig> cacheConfig;
+
+        public boolean isAnyPropertySet() {
+            return !cacheConfig.isEmpty();
+        }
+
+        public List<CacheConfigEntry> getCacheConfigurationEntries() {
+            List<CacheConfigEntry> cacheRegionsConfigEntries = new ArrayList<>();
+            for (Map.Entry<String, CacheConfig> regionEntry : cacheConfig.entrySet()) {
+                List<CacheConfigEntry> entries = regionEntry.getValue()
+                        .getCacheConfigEntries()
+                        .stream()
+                        .map(configEntry -> new CacheConfigEntry("hibernate.cache." + regionEntry.getKey() + configEntry.key,
+                                configEntry.value))
+                        .collect(Collectors.toList());
+
+                cacheRegionsConfigEntries.addAll(entries);
+            }
+
+            return cacheRegionsConfigEntries;
+        }
+    }
+
+    @ConfigGroup
+    public static class CacheConfig {
+        final static String EXPIRATION_MAX_IDLE = "expiration.max-idle";
+        final static String MEMORY_OBJECT_COUNT = "memory.object-count";
+
+        /**
+         * Cache expiration config
+         */
+        @ConfigItem
+        public ExpirationConfig expiration;
```

Comment:
My initial approach was to use this 
``` 
@ConfigItem(name = EXPIRATION_MAX_IDLE)
public OptionalLong expirationMaxIdle; 
```  
instead of having an expirationConfig object acting as a container. But this approach does not work. My observation was that a config property like this 
```
quarkus.hibernate-orm.cache."com.example.EntityB.expiration.max-idle"=86400
``` 
gave a hibernate property like 
```
hibernate.cache.expiration.expiration.max-idle"=86400
```

Reply:
Hmmm. I don't see how the map value would magically disappear if your code is adding it?

(I'm in favor of using names, that's why I ask)

Reply:
@gsmet thanks for taking a look.

> Hmmm. I don't see how the map value would magically disappear if your code is adding it?

It is not this current version. 

> (I'm in favor of using names, that's why I ask)

Me too. 

> My initial approach was to use this
> 
> ```
> @ConfigItem(name = EXPIRATION_MAX_IDLE)
> public OptionalLong expirationMaxIdle; 
> ```

Which seems to correspond what we both think is the best approach.  My goal is to arrive to a `CacheConfig` config group like below.

``` 
 @ConfigGroup
    public static class CacheConfig {
        private final static String EXPIRATION_MAX_IDLE = "expiration.max-idle";
        private final static String MEMORY_OBJECT_COUNT = "memory.object-count";

        /**
         * Cache expiration config
         */
        @ConfigItem(name = EXPIRATION_MAX_IDLE)
        public OptionalLong expirationMaxIdle;

        /**
         * Cache memory config
         */
        @ConfigItem(name = MEMORY_OBJECT_COUNT)
        public OptionalLong memoryObjectCount;
    } 
```

And a map config item in `HibernateOrmConfig`.
```
  /**
     * Caching configuration
     */
    public Map<String, CacheConfig> cache; 
```

I have not had much time to look further at this lately. So from my recent memory, the above configuration does not get `infinispan-cache-jpa` tests to pass. I did not debugged further but it seemed to me that keys of the `cache` are either `expiration` or `memory` instead of cache regions name: `com.example.EntityB` and `com.example.EntityA` respectively. 
With the code snippets, am I doing something wrong?

Reply:
Hmmm, right, for whatever reasons, it doesn't seem to work.

I don't think it's worth fighting this right now as we might have to add a couple more config knobs later anyway.

I made a few simplifications along the way of trying that, I will push an additional commit.

I also think we should change the `max-idle` one to be a `Duration`. I sent an email to the list to have some feedback on this as it would be a breaking change.

Reply:
@gsmet Thanks for taking a look.

> Hmmm, right, for whatever reasons, it doesn't seem to work.
> 
> I don't think it's worth fighting this right now as we might have to add a couple more config knobs later anyway.

Do you think it is worth opening an issue for?

> I made a few simplifications along the way of trying that, I will push an additional commit.

I quickly reviewed them and everything looks good to me. Better naming also,  thanks. 

> I also think we should change the `max-idle` one to be a `Duration`. I sent an email to the list to have some feedback on this as it would be a breaking change.

Okay, ping me if I can be helped once we have those feedback so that I can send the PR.